src/reporter.ts

```diff
@@ -2,10 +2,13 @@
 import { EnvMissingError } from './errors'
 import { ReporterOptions } from './types'
 
+type Errors<T> = Partial<Record<keyof T, Error>>;
+type Logger = (data: any, ...args: any[]) => void;
+
 // The default reporter is supports a second argument, for consumers
 // who want to use it with only small customizations
 type ExtraOptions<T> = {
-  onError?: (errors: Partial<Record<keyof T, Error>>) => void
+  onError?: (errors: Errors<T>) => void
   logger: (output: string) => void
```

Comment:
The signature of the `logger` arg here should be updated, no?

Reply:
It doesn't have to change.

This MR was more about allowing users that specify their own `errorReporter` like described [in the README](https://github.com/af/envalid#error-reporting) to use  the same logs formatting logic.
One would simply `import { envalidLogger } from "envalid";` and use it in his custom `errorReporter`. That is all.

If you mean that we should change the current `logger` interface to `envalidLogger`'s, that would be a breaking change.

I didn't know what parts of `ExtraOptions` were meant to be "user configuration". Since that wasn't specified in the README I just assumed it was purposefully omitted and not meant to be user configurable.
I suppose you could have gradual configuration for the whole errorReporting process:
- You can already specify an entirely different errorReporter.
- **You could add an option to specify the "errorLogger" (what I called the "envalidLogger")** (TODO?)
- And logger in your interface allows to control the "low levl" logging process, where the error messages are printed basically.

That was simply not the goal of that PR.